### PR TITLE
Use api go module for hashicorp/vault.

### PR DESCRIPTION
**Pull request by bradbl**
The Hashicorp Vault project  only supports imports of the api or sdk
modules since these were carved out from the main respository.
See https://github.com/hashicorp/vault/issues/7205.

Explicitly taking a dependency on the api module itself allows
this project to "play nice" with other respositories which take
either direct or transitive dependencies on it.

https://github.com/hashicorp/vault/issues/7848 contains a description
of how retaining an import of the root github.com/hashicorp/vault can
cause dependency resolution issues in downstream projects.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :x: [more...](#e9fee5b2-1ff4-45c4-a633-308c9457e747) |
| mint-large-bucket.sh | :x: [more...](#e7667243-5610-43b1-a9e9-ff73dff3becf) |
| mint-compression-dist-xl.sh | :x: [more...](#cf4cf8de-f1e7-4fa7-af1a-6ef1f81b5733) |
<a name="e9fee5b2-1ff4-45c4-a633-308c9457e747"></a>
### 8502-f0f4d38/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:32675
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 830555e6760a:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 5 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-ip4j21omlms3nu",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364958,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 15D597B3333C6CC8, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```
<a name="cf4cf8de-f1e7-4fa7-af1a-6ef1f81b5733"></a>
### 8502-f0f4d38/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:30618
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f547b446921f:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 7 seconds
(2/14) Running aws-sdk-java tests ... done in 7 seconds
(3/14) Running aws-sdk-php tests ... done in 1 minutes and 30 seconds
(4/14) Running aws-sdk-ruby tests ... done in 22 seconds
(5/14) Running awscli tests ... done in 3 minutes and 15 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 44 seconds
(8/14) Running minio-dotnet tests ... done in 2 minutes and 1 seconds
(9/14) Running minio-go tests ... done in 6 minutes and 4 seconds
(10/14) Running minio-java tests ... done in 12 minutes and 57 seconds
(11/14) Running minio-js tests ... done in 1 minutes and 51 seconds
(12/14) Running minio-py tests ... FAILED in 13 minutes and 24 seconds
{
  "duration": 151,
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1345, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2040, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1359, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "status": "FAIL",
  "args": {
    "object_name": "374a21dd-6ac0-476a-b017-60fc5d72616f",
    "presigned_get_object_url": "http://72.28.97.54:30618/minio-py-test-447f7dc6-1d57-4a19-907c-878470b67e7e/374a21dd-6ac0-476a-b017-60fc5d72616f?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20191109%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20191109T205117Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&X-Amz-Signature=d7464fe92b857c85bf846f0d93cc2bc07b5305912b8fdfe4ad41df9ee8b8ffa6",
    "response.status": 403,
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"440\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/xml\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2019-11-09T19-49-04Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"15D599BF4FBBBF86\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Sat, 09 Nov 2019 20:51:31 GMT\"]}}",
    "response._body": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDenied</Code><Message>Request has expired</Message><Key>374a21dd-6ac0-476a-b017-60fc5d72616f</Key><BucketName>minio-py-test-447f7dc6-1d57-4a19-907c-878470b67e7e</BucketName><Resource>/minio-py-test-447f7dc6-1d57-4a19-907c-878470b67e7e/374a21dd-6ac0-476a-b017-60fc5d72616f</Resource><RequestId>15D599BF4FBBBF86</RequestId><HostId>79129cd1-4cc9-42e6-80d8-2d1b89b40581</HostId></Error>",
    "bucket_name": "minio-py-test-447f7dc6-1d57-4a19-907c-878470b67e7e",
    "response.reason": "Forbidden"
  },
  "message": "AccessDenied: message: Access Denied",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "name": "minio-py:test_presigned_get_object_expiry_5sec"
}

Executed 11 out of 14 tests successfully.
```
<a name="e7667243-5610-43b1-a9e9-ff73dff3becf"></a>
### 8502-f0f4d38/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.58:31115
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp bb47550cb6ca:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 23 seconds
(2/14) Running aws-sdk-java tests ... done in 4 seconds
(3/14) Running aws-sdk-php tests ... done in 1 minutes and 56 seconds
(4/14) Running aws-sdk-ruby tests ... done in 1 minutes and 9 seconds
(5/14) Running awscli tests ... done in 3 minutes and 42 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 2 minutes and 18 seconds
(8/14) Running minio-dotnet tests ... done in 3 minutes and 22 seconds
(9/14) Running minio-go tests ... FAILED in 10 minutes and 43 seconds
{
  "args": {},
  "duration": 101194,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
Error: No such image: minio/minio:8502-f0f4d38

File: go.mod [hunk 1]
```diff
@@ -27,15 +27,15 @@ require (
 	github.com/gorilla/handlers v1.4.0
 	github.com/gorilla/mux v1.7.0
 	github.com/gorilla/rpc v1.2.0+incompatible
-	github.com/hashicorp/vault v1.1.0
+	github.com/hashicorp/vault/api v1.0.5-0.20190730042357-746c0b111519
```
Comment (aead) [inline go.mod]: Shouldn't that be `v1.0.5`?! (without date/commit-id)

Comment (bradbl) [reply go.mod]: Thanks, updated to resolve.

**Review by aead — APPROVED** [review]

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Can we limit the updates to go.mod to only vault? - we are upgrading other deps which I think should be avoided.

**Comment by bradbl** [comment]
> Can we limit the updates to go.mod to only vault? - we are upgrading other deps which I think should be avoided.

No, these other dependencies are automatically updated by `go mod tidy`.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#c915100b-8974-4579-9791-7c3d418f88a5) |
| mint-compression-dist-xl.sh | :x: [more...](#b0d355b3-381b-4c52-91a0-e628ba0fb7af) |
<a name="c915100b-8974-4579-9791-7c3d418f88a5"></a>
### 8502-dc22451/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.55:30837
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp ff086b13513c:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 8 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 1 minutes and 10 seconds
(4/14) Running aws-sdk-ruby tests ... done in 25 seconds
(5/14) Running awscli tests ... done in 1 minutes and 46 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 38 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 57 seconds
(9/14) Running minio-go tests ... FAILED in 6 minutes and 14 seconds
{
  "args": {},
  "duration": 136644,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
<a name="b0d355b3-381b-4c52-91a0-e628ba0fb7af"></a>
### 8502-dc22451/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:32009
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 7ad0f6034476:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 2 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 1 minutes and 3 seconds
(4/14) Running aws-sdk-ruby tests ... done in 9 seconds
(5/14) Running awscli tests ... done in 2 minutes and 10 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 19 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 4 seconds
(9/14) Running minio-go tests ... done in 2 minutes and 12 seconds
(10/14) Running minio-java tests ... done in 6 minutes and 52 seconds
(11/14) Running minio-js tests ... done in 1 minutes and 24 seconds
(12/14) Running minio-py tests ... FAILED in 8 minutes and 55 seconds
{
  "duration": 188,
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "message": "'content-language'",
  "status": "FAIL",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1400, in test_presigned_get_object_response_headers\n    returned_content_language = response.headers['Content-Language']\n  File \"/usr/local/lib/python3.5/dist-packages/urllib3/_collections.py\", line 156, in __getitem__\n    val = self._container[key.lower()]\nKeyError: 'content-language'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2043, in main\n    test_presigned_get_object_response_headers(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1418, in test_presigned_get_object_response_headers\n    raise Exception(err)\nException: 'content-language'\n",
  "name": "minio-py:test_presigned_get_object_response_headers",
  "args": {
    "bucket_name": "minio-py-test-a8ab3164-f3e9-4ede-a655-5af6dde9f6c5",
    "presigned_get_object_url": "http://72.28.97.52:32009/minio-py-test-a8ab3164-f3e9-4ede-a655-5af6dde9f6c5/dcc97816-2ead-474d-bb43-53d5dce16f3f?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20191111%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20191111T184038Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&response-content-language=en_US&response-content-type=text%2Fplain&X-Amz-Signature=02d813b75433e1d27d498248c98f70ca66614b55e961157eeca8d64df4c08a59",
    "object_name": "dcc97816-2ead-474d-bb43-53d5dce16f3f"
  }
}

Executed 11 out of 14 tests successfully.
```
Error: No such image: minio/minio:8502-dc22451

**Comment by harshavardhana** [comment]
Closing #8631 already fixes the vault import.

**Pull request closed without merging**